compiler/rustc_middle/src/ty/structural_impls.rs

```diff
@@ -162,16 +162,15 @@ impl<'tcx> fmt::Debug for ty::consts::Expr<'tcx> {
 impl<'tcx> fmt::Debug for ty::Const<'tcx> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         // If this is a value, we spend some effort to make it look nice.
-        if let ConstKind::Value(_, _) = self.kind() {
+        if let ConstKind::Value(_) = self.kind() {
             return ty::tls::with(move |tcx| {
-                // Somehow trying to lift the valtree results in lifetime errors, so we lift the
-                // entire constant.
+                // ValTrees aren't interned, so we lift the entire constant.
                 let lifted = tcx.lift(*self).unwrap();
-                let ConstKind::Value(ty, valtree) = lifted.kind() else {
+                let ConstKind::Value(cv) = lifted.kind() else {
                     bug!("we checked that this is a valtree")
                 };
                 let mut cx = FmtPrinter::new(tcx, Namespace::ValueNS);
-                cx.pretty_print_const_valtree(valtree, ty, /*print_ty*/ true)?;
+                cx.pretty_print_const_valtree(cv.valtree, cv.ty, /*print_ty*/ true)?;
```

Comment:
would it make sense to make `pretty_print_const_valtree` just be `pretty_print_const_value` and accept `ty::Value`?

Reply:
Now that you mention `valtree_to_const_value` I'm realising that there's an unfortunate naming overlap between `mir::ConstValue` and `ty::Const::Value` lol. We should probably just leave the name as `const_valtree` even if it's updated to take a `ty::Value` as otherwise you'd wind up with weird functions like `const_value_to_const_value`...

Reply:
for `valtree_to_pat` I'd just leave that up to oli as he's more familiar with the way mir pattern logic interacts with `ty::Const` and I don't want to make a decision there since I don't really work on that logic :thinking:

Reply:
Seems reasonable to me to use this in `valtree_to_pat`.

Reply:
Leave a FIXME for `pretty_print_const_valtree` to be changed to using this type